src/sparkpilot/api.py

```
@@ -408,10 +436,129 @@ class AccessContext:
+        separators=(",", ":"),
+        sort_keys=True,
+    ).encode("utf-8")
+    body_enc = _urlsafe_b64encode(body)
+    mac = hmac.new(
+        _state_signing_key(), body_enc.encode("ascii"), hashlib.sha256
+    ).digest()
+    sig_enc = _urlsafe_b64encode(mac)
+    return f"{_INVITE_STATE_PREFIX}.{body_enc}.{sig_enc}"
+
+
+def _decode_invite_state(state: str | None) -> InviteStatePayload | None:
```

Ruff (0.15.11)

[warning] Too many return statements (10 > 6)

(PLR0911)